src/backend_input_collector.cc

```diff
@@ -738,6 +739,8 @@ BackendInputCollector::BatchInputShape(
       }
       break;
     }
+    default:
```

Comment:
Should we not return an error here instead?

Reply:
Don't add `default` so there will be compile-time error if we have added new batch input kind but forget to add the handling code

Reply:
I agree with you but I am not sure.. Ideally we will never hit this logic as kind can take only the above 4 values for now. 
I just added the default here because that's what we do in core backend utils.
https://github.com/triton-inference-server/server/blob/main/src/core/backend_context.cc#L721

But I can return an error here if there was no other intention of returning with empty shape for future Kind values not covered with above cases.

Reply:
I will be removing the default as per Guan's comment.